src-tauri/src/lib.rs

```diff
@@ -212,18 +217,17 @@ pub fn run() {
                 // This is necessary to among other things stop the monero-wallet-rpc process
                 // If the application is forcibly closed, this may not be called.
                 // TODO: fix that
-                let context = app.state::<RwLock<State>>().inner().try_read();
-
-                match context {
-                    Ok(context) => {
-                        if let Some(context) = context.context.as_ref() {
-                            if let Err(err) = context.cleanup() {
-                                println!("Cleanup failed {}", err);
-                            }
-                        }
-                    }
-                    Err(err) => {
-                        println!("Failed to acquire lock on context: {}", err);
+                let state = app.state::<State>();
+                let context_to_cleanup = if let Ok(context_lock) = state.context.try_read() {
+                    context_lock.clone()
+                } else {
+                    println!("Failed to acquire lock on context");
+                    None
+                };
+
+                if let Some(context) = context_to_cleanup {
+                    if let Err(err) = context.cleanup() {
```

Comment:
@Matroskine Maybe we can call `store(...)` here on all wallets (in the cleanup function)